### PR TITLE
support lonelynode ha setup

### PR DESCRIPTION
**Pull request by jdsn**
proposed changes to allow ha setup consisting of lonelynodes only

Note: the step list needs to be changed from the default to this (maybe followed by other steps):
`mkcloud cleanup prepare setupadmin addupdaterepo runupdate prepareinstcrowbar bootstrapcrowbar instcrowbar setuplonelynodes crowbar_register setup_aliases proposal`

### CODE REVIEW

File: scripts/lib/mkcloud-libvirt.sh [hunk 1]
```diff
@@ -320,23 +320,46 @@ function libvirt_do_onhost_deploy_image()
     fi
 }
 
+# create the libvirt configuration of a node (compute, controller, storage, lonely)
+function libvirt_do_onhost_create_compute_config()
```
Comment (dirkmueller) [inline scripts/lib/mkcloud-libvirt.sh]: imho this should not be called _do_, as that indicates that a corresponding implementation in dirmaint backend is needed. I would name it _libvirt_create_compute_config.. or similar.

File: scripts/mkcloud [hunk 2]
```diff
@@ -670,38 +678,27 @@ function restartcloud
...
-            fi
-        fi
-
-        ${mkcloud_lib_dir}/libvirt/compute-config $cloud $i $macaddress\
-            $controller_raid_volumes $cephvolumenumber "$drbd_serial"\
-            $compute_node_memory $controller_node_memory $libvirt_type $vcpus\
-            $emulator $vdisk_dir 3 $nodenumbercontroller > /tmp/$cloud-node$i.xml
-        ${mkcloud_lib_dir}/libvirt/vm-start /tmp/$cloud-node$i.xml
+        ${mkclouddriver}_do_onhost_create_compute_config $i
+        ${mkcloud_lib_dir}/${mkclouddriver}/vm-start /tmp/$cloud-node$i.xml
```
Comment (dirkmueller) [inline scripts/mkcloud]: if we're supposed to have setupnodes now for the lonelynodes case, then this code needs to move into lib/mkcloud-libvirt.sh, as /vm-start does not exist for dirmaint.

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -2320,6 +2320,9 @@ function onadmin_crowbar_register
     local pubkey=`cat /root/.ssh/id_rsa.pub`
     ssh_password $crowbar_register_node_ip "mkdir -p /root/.ssh; echo '$pubkey' >> /root/.ssh/authorized_keys"
 
+    # uninstall cloud-init, its dependecies break the installation of openstack
+    ssh_password $crowbar_register_node_ip "zypper --non-interactive rm -u cloud-init"
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: huh? there shouldn't be an image that still has cloud-init installed..

Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: there is, and even if we cleaned it up, its more secure to try to remove it :)

File: scripts/mkcloud (hunk 2 above)
Comment (jdsn) [inline scripts/mkcloud]: No, setupnodes != setuplonelynodes
Just the creation of the libvirt xml is unified.

Comment (jdsn) [inline scripts/mkcloud]: but I am fine with moving it to lib/mkcloud-libvirt.sh

Comment (jdsn) [inline scripts/mkcloud]: btw. vm-start was in setupnodes already - I just kept it there

**Comment by aplanas** [comment]
If this commit requires the update of openstack-mkcloud.xml, check this first: https://github.com/SUSE-Cloud/automation/pull/1302

**Comment by jdsn** [comment]
> If this commit requires the update of openstack-mkcloud.xml, check this first: #1302

@aplanas: no it does not, same as PR #1317 did not require it. Both PRs only moved files around.

Anyhow the moving was split off of this PR into PR #1317. So its done now and I will rebase this PR.

**Comment by dirkmueller** [comment]
@jdsn rebase?

**Comment by dirkmueller** [comment]
@jdsn rebase?

**Pull request closed without merging**